### PR TITLE
Index representative_checksum too, for fingerprinted URLs

### PR DESCRIPTION
**Pull request by jrochkind**
Using the checksum in the URL for assets is nice, becuase it menas you can make
them infinitely cacheable (For instance for browsers), even if the image behind
a file_id changes (versions?).  I am doing this for DZI experiment, but it's
potentially applicable to all our image sources. 

But that means it's gotta be indexed, so you have it to generate the URLs
at display time. 

Yes, this apparently requires touching five different files!

### CODE REVIEW

File: app/presenters/chf/file_set_presenter.rb [hunk 1]
```diff
@@ -20,6 +20,12 @@ def representative_file_id
       end
     end
 
+    def representative_checksum
+      # I can not explain why this didn't work with a delegate to solr_document
```
Comment (hackmastera) [inline app/presenters/chf/file_set_presenter.rb]: When you delegate to solr_document you still have to write an accessor method in the solr_document model. To bridge to the solr_name.

**Comment by hackmastera** [comment]
Does this go along with a routing change? I will see more context in the dzi PR?

File: app/presenters/chf/file_set_presenter.rb (hunk 1 above)
Comment (jrochkind) [inline app/presenters/chf/file_set_presenter.rb]: yeah, I figured out what was going on eventually! AFter I committed so didn't make it into my comments, I should fix them. 

Man, this stack.

**Comment by jrochkind** [comment]
No routing change. In this case, it just changes the template for URLs in S3, those URLs are generated by ruby code in our app, Rails routing is not involved.

**Comment by jrochkind** [comment]
Using this approach in other tile-back-ends may or may not require routing changes. I'm not really thinking those out, just going wtih DZI for now. This PR puts the checksums in the index so they _can_ be more easily used for file URI's, that's it.

**Comment by hackmastera** [comment]
Got it, thanks!

File: app/presenters/chf/file_set_presenter.rb (hunk 1 above)
Comment (hackmastera) [inline app/presenters/chf/file_set_presenter.rb]: Cool! Yeah, probably should fix the comment. Otherwise +1 to merge; feel free to go ahead!

**Pull request merged**